Gems/AWSCore/Code/Include/Framework/ServiceRequestJob.h

```diff
@@ -658,11 +599,73 @@ namespace AWSCore
             AZ_Printf(logRequestsChannel, "Response Body:\n");
             PrintRequestOutput(responseContent);
         }
+
     public:
+        using OnSuccessFunction = AZStd::function<void(ServiceRequestJob* job)>;
+        using OnFailureFunction = AZStd::function<void(ServiceRequestJob* job)>;
+
         template<class Allocator = AZ::SystemAllocator>
         static ServiceRequestJob* Create(OnSuccessFunction onSuccess, OnFailureFunction onFailure = OnFailureFunction{}, IConfig* config = GetDefaultConfig())
         {
-            return azcreate(ServiceRequestJob, (onSuccess, onFailure, config), Allocator);
+            return azcreate(Function, (onSuccess, onFailure, config), Allocator);
+        }
+
+    };
+
+    /// A derived class that calls lambda functions on job completion.
+    template<class RequestTraits>
+    class ServiceRequestJob<RequestTraits>::Function
+        : public ServiceRequestJob
+    {
+    public:
+        // To use a different allocator, extend this class and use this macro.
+        AZ_CLASS_ALLOCATOR(Function, AZ::SystemAllocator, 0);
+
+        Function(OnSuccessFunction onSuccess, OnFailureFunction onFailure = OnFailureFunction{}, IConfig* config = GetDefaultConfig())
+            : ServiceRequestJob(false, config) // No auto delete - we'll take care of it
+            , m_onSuccess{ onSuccess }
+            , m_onFailure{ onFailure }
+        {
+        }
+
+    protected:
+        OnSuccessFunction m_onSuccess;
```

Comment:
Might be worth adding comment about why these are protected and not private? 

Same comment that methods before member variables